### PR TITLE
feat: support barrel files for ts and js [sc-0]

### PR DESCRIPTION
**Pull request by umutuzgur**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

### Notes for the reviewer
Support barrel files for TS and JS

### CODE REVIEW

File: packages/cli/src/services/check-parser/parser.ts [hunk 1]
```diff
@@ -88,6 +106,11 @@ export class Parser {
     // We can add a not-null assertion operator (!).
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       const item = bfsQueue.shift()!
+
+      if (item.filePath.endsWith(FileExtensions.PACKAGE)) {
+        // Hold info about the main file and doesn't need to be parser
```
Comment (clample) [inline packages/cli/src/services/check-parser/parser.ts:111]:
```suggestion
        // Hold info about the main file and doesn't need to be parsed
```

File: packages/cli/src/services/check-parser/parser.ts [hunk 2]
```diff
@@ -125,22 +148,35 @@ export class Parser {
...
+        deps.push({ filePath: fullPath, content })
+        if (extension === FileExtensions.PACKAGE) {
+          const { main } = JSON.parse(content)
+          if (!main || !main.length) {
+            // No main is defined. This means package.json doesn't have a specific entry
+            continue
+          }
+          const mainFile = path.join(filePath, main)
+          deps.push({
+            filePath: mainFile, content: fs.readFileSync(mainFile, { encoding: 'utf-8' }),
```
Comment (clample) [inline packages/cli/src/services/check-parser/parser.ts:176]: Do we need to automatically import the `main` file from a `package.json`?

Comment (umutuzgur) [reply packages/cli/src/services/check-parser/parser.ts:176]: Yeah. For example with the cli, it is `"main": "dist/index.js"`. If main doesn't exist, you go for `index.js` directly

Comment (clample) [reply packages/cli/src/services/check-parser/parser.ts:176]: I'm still confused, though.  In a check, what's the advantage of importing `package.json` and having the `main` automatically imported as well?

Comment (umutuzgur) [reply packages/cli/src/services/check-parser/parser.ts:176]: If we don't have the package.json, `import './module'` will point to `module/index.js`. If we have the package.json in the runtime, it should point to wherever the package.json specifies

Comment (clample) [reply packages/cli/src/services/check-parser/parser.ts:176]: Ah got it. Thank you for clarifying.

**Review by clample — APPROVED** [review]
Looks nice!

**Pull request merged**